### PR TITLE
feature: add validation for ads local.

### PR DESCRIPTION
**Pull request by nistormihai**

### CODE REVIEW

**Comment by vladnicoara** [comment]
LGTM

File: client/app/scripts/liveblog-edit/directives.js [hunk 1]
```diff
@@ -534,15 +541,44 @@ define([
...
         }])
-        .directive('freetypeLink', ['$compile', function($compile) {
+        .directive('freetypeText', ['$templateCache', function($templateCache) {
 
             return {
                 restrict: 'E',
-                template: '<input type="url" ng-model="link"/>',
-                controller: function() {
-                },
+                template: $templateCache.get('scripts/liveblog-edit/views/freetype-text.html'),
```
Comment (m0g) [inline client/app/scripts/liveblog-edit/directives.js]: What is this `$templateCache` used for?

File: client/app/scripts/liveblog-edit/directives.js [hunk 2]
```diff
@@ -534,15 +541,44 @@ define([
...
+                controller: ['$scope', function($scope) {
+                    $scope.valid = true;
+                    $scope._id = _.uniqueId('text');
+                    if ($scope.compulsory !== undefined) {
+                        $scope.$watch('[text,compulsory]', function(value) {
```
Comment (m0g) [inline client/app/scripts/liveblog-edit/directives.js]: What is this `[text,compulsory]` doing? Is it watch two different variables?

File: client/app/scripts/liveblog-edit/freetype.service.js [hunk 3]
```diff
@@ -166,7 +167,7 @@ function(angular) {
                     });
                     if (name) {
                         path2obj(scope[SCOPE_FREETYPEDATA], name);
-                        return '<input ng-model=' + makeAngularAttr(name, attr) + '/>';
+                        return '<freetype-text text=' + makeAngularAttr(name, attr) + ' validation="validation"></freetype-text>';
```
Comment (m0g) [inline client/app/scripts/liveblog-edit/freetype.service.js]: Injecting a directive from a string variable defined in a service? Doesn't sound right to me

File: client/app/scripts/liveblog-edit/directives.js (hunk 2 above)
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/directives.js]: yes

File: client/app/scripts/liveblog-edit/freetype.service.js (hunk 3 above)
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/freetype.service.js]: `transform` a $dollar template into an angular template.

File: client/app/scripts/liveblog-edit/directives.js (hunk 1 above)
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/directives.js]: `html`s files are feed to `$templateCache` via `grunt ngtemplates`

**Pull request closed without merging**